packages/php-wasm/universal/src/lib/base-php.ts

```diff
@@ -616,6 +617,77 @@ export abstract class BasePHP implements IsomorphicLocalPHP {
 		this[__private__dont__use].FS.rename(fromPath, toPath);
 	}
 
+	/** @inheritDoc */
+	@rethrowFileSystemError('Could not copy "{path}"')
+	cp(
+		sourcePath: string,
+		destinationPath: string,
+		options: CpOptions = { recursive: false }
+	) {
+		const FS = this[__private__dont__use].FS;
+
+		const sourceStat = FS.stat(sourcePath);
+
+		// The FILEMODE tells us things about the file, like
+		// permissions, whether its a file, link, or directory
+		// as well as its access and exec permissions.
+		//
+		// The 15th bit of the FILEMODE is the sticky-bit.
+		// Emscripten directories will have the sticky-bit set.
+		//
+		// We'll use a binary literal (0b...) with a logical
+		// and (&) to check for that.
+		const stickyBit = 0b100000000000000;
+		const sourceIsDir = sourceStat.mode & stickyBit;
+
+		let destinationExists: boolean;
+		let destinationIsDir = false;
+
+		try {
+			const destinationStat = FS.stat(destinationPath);
+			destinationExists = true;
+			// Similar check to above but testing if DESTINATION is a directory.
+			destinationIsDir = !!(destinationStat.mode & stickyBit);
+		} catch {
+			destinationExists = false;
+		}
+
+		if (!sourceIsDir) {
```

Comment:
What will happen if I copy a directory tree like `wordpress/wp-content/plugins` onto another directory, say `/www` that already has `wordpress/wp-content/plugins` in it? It seems like it would duplicate some directories?